src/graphnet/data/extractors/i3genericextractor.py

```diff
@@ -0,0 +1,205 @@
+from typing import Any, Dict, List, Optional, Union
+
+from graphnet.data.extractors.i3extractor import I3Extractor
+from graphnet.data.extractors.utilities.types import (
+    cast_object_to_pure_python,
+    cast_pulse_series_to_pure_python,
+)
+from graphnet.data.extractors.utilities.collections import (
+    transpose_list_of_dicts,
+    serialise,
+    flatten_nested_dictionary,
+)
+
+from graphnet.utilities.logging import get_logger
+
+logger = get_logger()
+
+try:
+    from icecube import (
+        dataclasses,
+        icetray,
+    )  # pyright: reportMissingImports=false
+except ImportError:
+    logger.warning("icecube package not available.")
+
+
+GENERIC_EXTRACTOR_NAME = "<GENERIC>"
+
+
+class I3GenericExtractor(I3Extractor):
+    """Dynamically and generically extract all information from frames."""
+
+    def __init__(
+        self,
+        keys: Optional[Union[str, List[str]]] = None,
+        exclude_keys: Optional[Union[str, List[str]]] = None,
+    ):
+        """Constructor."""
+        # Check(s)
+        if (keys is not None) and (exclude_keys is not None):
+            raise ValueError(
+                "Only one of `keys` and `exclude_keys` should be set."
+            )
+
+        # Cast(s)
+        if isinstance(keys, str):
+            keys = [keys]
+        if isinstance(exclude_keys, str):
+            exclude_keys = [exclude_keys]
+
+        # Reference to frame currently being processed
+        self._keys: List[str] = keys
+        self._exclude_keys: List[str] = exclude_keys
+
+        super().__init__(GENERIC_EXTRACTOR_NAME)
+
+    def _get_keys(self, frame: "icetray.I3Frame") -> List[str]:
+        """Get the effective list of keys to be queried from `frame`."""
+        if self._keys is None:
+            keys = list(frame.keys())
+            if self._exclude_keys is not None:
+                keys = [key for key in keys if key not in self._exclude_keys]
+        else:
+            keys = self._keys
+        return keys
+
+    def __call__(self, frame: "icetray.I3Frame") -> dict:
+        """Extract all possible data from `frame`."""
```

Comment:
If some fields are not extractable by this class, perhaps repeating them here is a good idea.